src/pages/api/user/update-username.ts

```diff
@@ -0,0 +1,46 @@
+import { PrismaClient } from '@prisma/client';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const prisma = new PrismaClient();
+
+const updateUsernameHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  const { email, username } = req.body as {
+    email: string;
+    username: string;
+  };
+
+  if (!email || !username) {
+    return res.status(400).json({
+      message: 'Please provide an email and username.'
+    });
+  }
+
+  try {
+    const duplicateUser = await prisma.user.findUnique({
+      where: {
+        name: username
+      }
+    });
+
+    if (duplicateUser) {
+      return res.status(400).json({
+        message: 'Username already exists.'
```

Comment:
Yeah idk if we need this though - or did you do this because the stats list usernames and we need to be able to tell who's who?